src/libraries/System.Text.Json/src/System/Text/Json/Writer/Utf8JsonWriter.cs

```diff
@@ -1004,7 +1004,10 @@ private void Grow(int requiredSize)
             {
                 Debug.Assert(_arrayBufferWriter != null);
 
-                _memory = _arrayBufferWriter.GetMemory(checked(BytesPending + sizeHint));
+                int needed = BytesPending + sizeHint;
```

Comment:
This was changed to keep the OOM exception semantics (instead of OverflowException). There are no effective perf differences since the overhead of the new `if` statement roughly equals the extra logic in using `checked()`.

Note there are other cases in STJ that also use `checked()` that could be changed like this code, but it doesn't use ABW so not changed in this PR. I may create a new issue to track this if deemed useful.

Reply:
> This was changed to keep the OOM exception semantics (instead of OverflowException). 

Add a test, if one doesn't exist?

Reply:
Yes the existing test that was modified in this PR was failing due to this (not throwing OOM as expected).